cmd/kubeadm/app/util/system/kernel_validator.go

```diff
@@ -233,34 +235,31 @@ func (k *KernelValidator) getKernelConfigReader() (io.Reader, error) {
 }
 
 // getKernelConfig gets kernel config from kernel config file and convert kernel config to internal type.
-func (k *KernelValidator) getKernelConfig() (map[string]kConfigOption, error) {
+func (k *KernelValidator) getKernelConfig() (map[string]kConfigOption, error, error) {
 	r, err := k.getKernelConfigReader()
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	return k.parseKernelConfig(r)
 }
 
 // parseKernelConfig converts kernel config to internal type.
-func (k *KernelValidator) parseKernelConfig(r io.Reader) (map[string]kConfigOption, error) {
+func (k *KernelValidator) parseKernelConfig(r io.Reader) (map[string]kConfigOption, error, error) {
 	config := map[string]kConfigOption{}
 	regex := regexp.MustCompile(validKConfigRegex)
 	s := bufio.NewScanner(r)
+	var errs []error
 	for s.Scan() {
 		if err := s.Err(); err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		line := strings.TrimSpace(s.Text())
 		if !regex.MatchString(line) {
 			continue
 		}
 		fields := strings.Split(line, "=")
-		if len(fields) != 2 {
-			klog.Errorf("Unexpected fields number in config %q", line)
-			continue
-		}
-		config[fields[0]] = kConfigOption(fields[1])
+		config[fields[0]] = kConfigOption(fields[1][0])
 	}
-	return config, nil
+	return config, errorsutil.NewAggregate(errs), nil
```

Comment:
To clarify my previous comment, we can remove the warning return here and from the other funcs, since we always return nothing (we never append anything to `errs`).